torchgeo/datasets/geo.py

```diff
@@ -427,10 +430,14 @@ def __getitem__(self, query: BoundingBox) -> Dict[str, Any]:
             data = self._merge_files(filepaths, query, self.band_indexes)
 
         sample = {"crs": self.crs, "bbox": query}
+
+        if self.dtype is not None:
+            data = data.to(self.dtype)
+
         if self.is_image:
             sample["image"] = data.float()
```

Comment:
We should remove this cast too for consistency.

Reply:
all images should be float though

Reply:
let's see what happens though :)

Reply:
It looks like this assumption is built into RasterDataset -- https://github.com/microsoft/torchgeo/blob/main/tests/datasets/test_geo.py#L201

There is a test that has a custom dataset with uint32 dtype that checks to make sure it has been converted by RasterDataset to float

Reply:
We can remove that test if we don't plan on unifying dtypes across datasets.

Reply:
I think it makes sense to cast images to floats because that is what will be passed to a network

Reply:
And I thought it would make sense to cast masks to long because that is what will be passed to torchmetrics 😆 

I think if a user specifies `dtype=torch.float16` when constructing the dataset and we instead use `torch.float32` just because it's an image, they'll be quite confused and will open a bug report. We should be consistent.